inc/event_target.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <map>
+#include <vector>
+#include "event_listener.hpp"
+
+namespace dragonslave {
+
+
+class EventBase
+{
+};
+
+
+class EventTarget
```

Comment:
EventTarget should only emit a single type of event. I think it's safe to put the templating here.